cluster/src/main/scala/org/apache/spark/memory/SnappyUnifiedMemoryManager.scala

```diff
@@ -635,6 +683,26 @@ class SnappyUnifiedMemoryManager private[memory](
     (offHeapStorageMemoryPool.memoryUsed > (maxOffHeapStorageSize * 0.90) ) ||
         (onHeapStorageMemoryPool.memoryUsed > (maxHeapStorageSize * 0.90))
   }
+
+  override def initMemoryStats(stats: MemoryManagerStats): Unit = {
+    stats.incMaxStorageSize(true, maxOffHeapStorageSize)
+    stats.incMaxStorageSize(false, maxHeapStorageSize)
+
+    stats.incStoragePoolSize(true, offHeapStorageMemoryPool.poolSize)
+    stats.incStoragePoolSize(false, onHeapStorageRegionSize)
+    stats.incStorageMemoryUsed(true, offHeapStorageMemoryPool.memoryUsed)
+    stats.incStorageMemoryUsed(false, onHeapStorageMemoryPool.memoryUsed)
+
+    stats.incExecutionPoolSize(true, offHeapExecutionMemoryPool.poolSize)
+    stats.incExecutionPoolSize(false, onHeapExecutionMemoryPool.poolSize)
+    stats.incExecutionMemoryUsed(true, offHeapExecutionMemoryPool.memoryUsed)
```

Comment:
@hbhanawat This looks like a design flaw in UMM itself. When executor goes down and comes up again it will create a new UMM and its pools sizes will start from 0. This will create a problem and soon we will hit OOM exception.

Reply:
@rishitesh as discussed, may be you can go ahead with the merge of this PR and handle the design issue as a separate task. This PR is important from debugging perspective.